specs/eventing/data-plane.md

```diff
@@ -1,125 +1,184 @@
+| ------------- | ------------------------------------------------- | ----- | ------------------ | ----- |
+| `1xx`         | (Unspecified)                                     | No\*  | No\*               | Yes\* |
+| `200`         | [Accepted, event in reply](#derived-reply-events) | No    | Yes                | No    |
+| `202`         | Event accepted                                    | No    | Yes                | No    |
+| other `2xx`   | (Unspecified)                                     | No    | Yes                | No    |
+| `3xx`         | (Unspecified)                                     | No\*  | No\*               | Yes\* |
+| `400`         | Unparsable event                                  | No    | No                 | Yes   |
+| `404`         | Endpoint does not exist                           | Yes   | No                 | Yes   |
+| `409`         | Conflict / Processing in progress                 | Yes   | No                 | Yes   |
+| `429`         | Too Many Requests / Overloaded                    | Yes   | No                 | Yes   |
+| other `4xx`   | Error                                             | No    | No                 | Yes   |
+| `5xx`         | Error                                             | Yes   | No                 | Yes   |
+
+\* Unspecified `1xx`, `2xx`, and `3xx` response codes are **reserved for future
+extension**. Event recipients SHOULD NOT send these response codes in this spec
+version, but event senders MUST handle these response codes as errors or success
+as appropriate and implement described success or failure behavior.
+
+Recipients MUST accept duplicate delivery of events, but they are not REQUIRED
+to detect that they are duplicates. If duplicate detection is implemented, then
+as specified in the
+[CloudEvents specification](https://github.com/cloudevents/spec/blob/v1.0.1/primer.md#id),
+event recipients MUST use the
+[`source` and `id` attributes](https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#required-attributes)
+to identify duplicate events. This specification does not describe state
+requirements for recipients which need to detect duplicate events. In general,
+senders MAY add or update other CloudEvent attributes on each delivery attempt;
+see [observability](#observability) for an example case.
+
+Where possible, event senders SHOULD re-attempt delivery of events where the
+HTTP request returned a retryable status code. It is RECOMMENDED that event
+senders implement some form of congestion control (such as exponential backoff)
+and delivery throttling when managing retry timing. Congestion control MAY cause
+event delivery to fail or MAY include not retrying failed delivery attempts.
+This specification does not document any specific congestion control algorithm
+or parameters. [Brokers](./overview.md#broker) and
+[Channels](./overview.md#channel) MUST implement congestion control and MUST
+implement retries.
```

Comment:
are we adding retries in this case for things like InMemoryChannel?

Reply:
I believe InMemoryChannel already does retries (at least as long as the pods backing the channel is not restarted).

If not, we'd need to fix it (which seems right to me).